### PR TITLE
fix: [FC-2] issue receive message in the past make duplicate

### PR DESCRIPTION
**Pull request by huydosgtech**

### CODE REVIEW

File: src/chat/ChatScreen.tsx [hunk 1]
```diff
@@ -130,10 +130,15 @@ export const ChatScreen: React.FC<ChatScreenProps> = ({
 
   useEffect(() => {
     let receiveMessageRef: () => void;
+    const curerntTime = Date.now();
```
Comment (loc-nguyenthien) [inline src/chat/ChatScreen.tsx:133]:
```suggestion
    const currentTime = Date.now();
```

File: src/chat/ChatScreen.tsx [hunk 2]
```diff
@@ -130,10 +130,15 @@ export const ChatScreen: React.FC<ChatScreenProps> = ({
...
     if (conversationRef.current?.id) {
       receiveMessageRef = firebaseInstance.receiveMessageListener(
         (message: MessageProps) => {
-          if (userInfo && message.senderId !== userInfo.id) {
+          if (
+            userInfo &&
+            message.senderId !== userInfo.id &&
+            message.createdAt >= new Date(curerntTime)
```
Comment (loc-nguyenthien) [inline src/chat/ChatScreen.tsx:140]:
```suggestion
            message.createdAt >= new Date(currentTime)
```

**Review by loc-nguyenthien — CHANGES_REQUESTED** [review]

File: src/chat/ChatScreen.tsx (hunk 1 above)
Comment (huydosgtech) [reply src/chat/ChatScreen.tsx]: ok I changed it

File: src/chat/ChatScreen.tsx (hunk 2 above)
Comment (huydosgtech) [reply src/chat/ChatScreen.tsx]: I changed. Pls check again.

**Review by baonguyenhsts — APPROVED** [review]

**Review by danhmaisgt — APPROVED** [review]

File: src/chat/ChatScreen.tsx [hunk 3]
```diff
@@ -130,10 +130,15 @@ export const ChatScreen: React.FC<ChatScreenProps> = ({
 
   useEffect(() => {
     let receiveMessageRef: () => void;
+    const currentTime = Date.now();
```
Comment (honghoangsts) [inline src/chat/ChatScreen.tsx:133]: I think we need convert to UTC

**Review by honghoangsts — CHANGES_REQUESTED** [review]

File: src/services/firebase/firestore.ts [hunk 4]
```diff
@@ -345,8 +346,13 @@ export class FirestoreServices {
       .onSnapshot((snapshot) => {
```
Comment (honghoangsts) [inline src/services/firebase/firestore.ts:346]: .where('createAt', '>', getCurrentTimestamp())
.onSnapshot(...)

**Review by honghoangsts — APPROVED** [review]

**Pull request merged**